### PR TITLE
Asymmetric signatures

### PR DESCRIPTION
**Pull request by rampantmonkey**

### CODE REVIEW

File: signature.js [hunk 1]
```diff
@@ -1,35 +1,54 @@
...
+    verify: buildVerify(crypto.createPublicKey(publicKey)),
+    publicKey: publicKey
+  }
 }
 
-module.exports = {
-  generate: generate,
-  verify: verify
+module.exports = (options = {}) => {
+  if(options.privateKey && options.publicKey) {
```
Comment (4kategreen) [inline signature.js]: will there be error handling should the public and private keys not exist?

File: signature.js [hunk 2]
```diff
@@ -1,35 +1,54 @@
...
-const verify = (token) => {
-  let components = token.split('.', 3)
-  if(components.length !== 3) { return false }
-  let msg = components[0] + '.' + components[1]
-  let providedSignature = components[2]
-  let computedSignature = sign(msg)
-  return providedSignature === computedSignature
+const build = (privateKey, publicKey) => {
+  return {
+    generate: buildGenerate(buildSign(crypto.createPrivateKey(privateKey))),
```
Comment (4kategreen) [inline signature.js]: why are you nesting these functions in this way? i don't see `buildSign` being used elsewhere? is there a more efficient way to do this that doesn't involve mental gymnastics?

File: signature.js (hunk 1 above)
Comment (rampantmonkey) [reply signature.js]: It "handles" the error by not producing the signature service. And then leaves it up to the caller what to do next. We will probably want a more standard convention once we have more than one service (or more than one use of this service).

File: signature.js (hunk 2 above)
Comment (rampantmonkey) [reply signature.js]: I wrote a function for each step of the verification process. If that function requires runtime information (like a key) there is a wrapper `build*` that creates the function with the argument pre-populated. The goal is to use [currying](https://en.wikipedia.org/wiki/Currying) to make it clear which arguments are intended as "user facing" and which are implementation details (or just dealing with javascript variable scoping).

**Pull request merged**